kinvolk/inspektor-gadget
# Fix DeleteTrace (fd leak and go-routine leak)

# Fix DeleteTrace (fd leak and go-routine leak)

Gadgets' DeleteTrace() functions were never called. This can cause fd leaks and go routine leaks.

This patch fixes the issue.

## How to use

No changes for users.

## Testing done

TODO:

- Use gadgets such as dns, network-policy, seccomp or biolatency which have a DeleteTrace function
- Delete the Trace custom resource without stopping it
- Check the file descriptor list of thegadgettracermanager process (`/proc/$pid/fd`): the bpf program and maps for the gadget should have been removed
- Check the go routine list of thegadgettracermanager process (`gadgettracermanager -dump`): go routines for gadgets should not be present anymore

Unit tests were added in the local-gadget PR.
